### ISSUE TITLE
Unable to parse shared mutation data

### ISSUE THREAD

**Issue by justerror** [xanf/vuex-shared-mutations#13]
OS: macOs 10.14.1
Bug reproduced in: 
- Chrome 70.0.3538.102 (Official Build) (64-bit)
- Safari 12.0.1 (14606.2.104.1.1)
- Opera 56.0.3051.52

Not reproduced in:
- Firefox 62.0.3 (64-bit);

I'm getting errors with last version of package (0.0.5). 

```
vuex-shared-mutations.js:67 [vuex-shared-mutations] Unable to parse shared mutation data
vuex-shared-mutations.js:68 notification TypeError: Cannot read property 'type' of null at eval
```

**Comment by justerror** [comment]
It seems that the 'notification' record from localstorage is already deleted when 'storage' the event is emitted. And this line give us `null`:
`var mutation = JSON.parse(window.localStorage.getItem(storageKeyEntry));`

**Comment by mathijswesterhof** [comment]
placing a timeout around the localstorage remove code can solve this error, **don't take this as a final solution though**.
in the vuex-shared-mutations.js > line 49
`
setTimeout(() => {
    window.localStorage.removeItem(key);
    window.localStorage.removeItem(storageKeyEntry);
},50)
`

**Comment by pranabpadhy** [comment]
**if (event.key !== key) return;**
update above line to **if (event.key !== storageKeyEntry) return;**

**var mutation = JSON.parse(window.localStorage.getItem(storageKeyEntry));**
update above line to  **var mutation = JSON.parse(event.newValue);**

these two changes fixed the issue for me.

**Comment by hanjeahwan** [comment]
> **if (event.key !== key) return;**
> update above line to **if (event.key !== storageKeyEntry) return;**
> 
> **var mutation = JSON.parse(window.localStorage.getItem(storageKeyEntry));**
> update above line to **var mutation = JSON.parse(event.newValue);**
> 
> these two changes fixed the issue for me.

Can you create this pull request

**Comment by pranabpadhy** [comment]
@hanjeahwan Done. Depends on @xanf to merge it.

**Comment by kenanyildiz** [comment]
> **if (event.key !== key) return;**
> update above line to **if (event.key !== storageKeyEntry) return;**
> 
> **var mutation = JSON.parse(window.localStorage.getItem(storageKeyEntry));**
> update above line to **var mutation = JSON.parse(event.newValue);**
> 
> these two changes fixed the issue for me.

When i apply this i am not getting error. But mutation does not shared, tabs are not sync. 😢

**Comment by xanf** [comment]
Hmm.... That's weird. Thank you for your report, I'll take a look ASAP

**Comment by uniibu** [comment]
> **if (event.key !== key) return;**
> update above line to **if (event.key !== storageKeyEntry) return;**
> 
> **var mutation = JSON.parse(window.localStorage.getItem(storageKeyEntry));**
> update above line to **var mutation = JSON.parse(event.newValue);**
> 
> these two changes fixed the issue for me.

As stated above, this indeed removes the error message, and syncs the localStorage but does not trigger the mutation on other tabs/windows.  This removes the exact purpose of this repo.

My workaround is to simply move `window.localStorage.removeItem(key);` and `window.localStorage.removeItem(storageKeyEntry);` at the `finally`

```
...
} finally {
   window.localStorage.removeItem(key);
   window.localStorage.removeItem(storageKeyEntry);
   committing = false;
}
```

**Comment by xanf** [comment]
I've fixed this issue in `0.0.7` but this is actually still incomplete solution. expect complete solution by 1.0.0 soon
